x/concentrated-liquidity/msg_server_test.go

```diff
@@ -202,18 +224,45 @@ func (suite *KeeperTestSuite) TestCollectIncentives_Events() {
 	testcases := map[string]struct {
 		upperTick                      int64
 		lowerTick                      int64
-		positionId                     uint64
+		positionIds                    []uint64
+		numPositionsToCreate           int
 		expectedCollectIncentivesEvent int
 		expectedMessageEvents          int
 		expectedError                  error
 		errorFromValidateBasic         error
 	}{
-		"happy path": {
+		"single position ID": {
 			upperTick:                      DefaultUpperTick,
 			lowerTick:                      DefaultLowerTick,
-			positionId:                     DefaultPositionId,
+			positionIds:                    []uint64{DefaultPositionId},
+			numPositionsToCreate:           1,
 			expectedCollectIncentivesEvent: 1,
-			expectedMessageEvents:          2, // 1 for collect incentives, 1 for message
+			expectedMessageEvents:          2, // 1 for collect incentives, 1 for send message
+		},
+		"two position IDs": {
+			upperTick:                      DefaultUpperTick,
+			lowerTick:                      DefaultLowerTick,
+			positionIds:                    []uint64{DefaultPositionId, DefaultPositionId + 1},
+			numPositionsToCreate:           2,
+			expectedCollectIncentivesEvent: 1,
+			expectedMessageEvents:          3, // 1 for collect incentives, 2 for send messages
+		},
+		"three position IDs": {
+			upperTick:                      DefaultUpperTick,
+			lowerTick:                      DefaultLowerTick,
+			positionIds:                    []uint64{DefaultPositionId, DefaultPositionId + 1, DefaultPositionId + 2},
+			numPositionsToCreate:           3,
+			expectedCollectIncentivesEvent: 1,
+			expectedMessageEvents:          4, // 1 for collect incentives, 3 for send messages
+		},
+		"error": {
+			upperTick:                      DefaultUpperTick,
+			lowerTick:                      DefaultLowerTick,
+			positionIds:                    []uint64{DefaultPositionId, DefaultPositionId + 1, DefaultPositionId + 2},
+			numPositionsToCreate:           2,
+			expectedCollectIncentivesEvent: 0,
+			expectedMessageEvents:          2, // 2 emitted for send messages
```

Comment:
Same question on this

Reply:
Replied in another thread. I think this is expected